cairo/ethereum/cancun/vm/instructions/system.cairo

```diff
@@ -125,3 +164,185 @@ func return_{range_check_ptr, evm: Evm}() -> EthereumException* {
     let ok = cast(0, EthereumException*);
     return ok;
 }
+
+func generic_create{
+    process_create_message_label: felt*,
+    range_check_ptr,
+    bitwise_ptr: BitwiseBuiltin*,
+    keccak_ptr: KeccakBuiltin*,
+    poseidon_ptr: PoseidonBuiltin*,
+    evm: Evm,
+}(
+    endowment: U256,
+    contract_address: Address,
+    memory_start_position: U256,
+    memory_size: U256,
+    init_code_gas: Uint,
+) -> EthereumException* {
+    alloc_locals;
+
+    let fp_and_pc = get_fp_and_pc();
+    local __fp__: felt* = fp_and_pc.fp_val;
+
+    let memory = evm.value.memory;
+    let call_data = memory_read_bytes{memory=memory}(memory_start_position, memory_size);
+    EvmImpl.set_memory(memory);
+
+    let is_gt_two_max_code_size = is_le(2 * MAX_CODE_SIZE + 1, call_data.value.len);
```

Comment:
doing `let is_calldata_within_bounds = is_le(call_data.value.len, 2 * MAX_CODE_SIZE + 1)`

and checking whether is_calldata_within_bounds == 0 would allow you avoid doing +1 and as such risk overflowing if MAX_CODE_SIZE > P

but even M31 >>>>> MAX_CODE_SIZE for the foreseeable future
so ignore this
but to keep in mind if we want to avoid incrementing felt conditions